packages/ra-data-simple-rest/src/index.ts

```diff
@@ -37,156 +37,169 @@ export default (
+                if (!headers.has(countHeader)) {
+                    throw new Error(
+                        `The ${countHeader} header is missing in the HTTP Response. The simple REST data provider expects responses for lists of resources to contain this header with the total number of results to build the pagination. If you are using CORS, did you declare ${countHeader} in the Access-Control-Expose-Headers header?`
+                    );
+                }
+                return {
+                    data: json,
+                    total:
+                        countHeader === 'Content-Range'
+                            ? parseInt(
+                                  headers.get('content-range').split('/').pop(),
+                                  10
+                              )
+                            : parseInt(headers.get(countHeader.toLowerCase())),
+                };
+            });
+        },
+
+        update: (resource, params) =>
+            httpClient(`${apiUrl}/${resource}/${params.id}`, {
+                method: 'PUT',
+                body: JSON.stringify(params.data),
+            }).then(({ json }) => ({ data: json })),
+
+        // simple-rest doesn't handle provide an updateMany route, so we fallback to calling update n times instead
+        updateMany: (resource, params) =>
+            Promise.all(
+                params.ids.map(id =>
+                    httpClient(`${apiUrl}/${resource}/${id}`, {
+                        method: 'PUT',
+                        body: JSON.stringify(params.data),
+                    })
+                )
+            ).then(responses => ({
+                data: responses.map(({ json }) => json.id),
+            })),
+
+        create: (resource, params) =>
+            httpClient(`${apiUrl}/${resource}`, {
+                method: 'POST',
+                body: JSON.stringify(params.data),
+            }).then(({ json }) => ({ data: json })),
+
+        delete: (resource, params) =>
+            httpClient(`${apiUrl}/${resource}/${params.id}`, {
+                method: 'DELETE',
+                headers: new Headers({
+                    'Content-Type': 'text/plain',
+                }),
+            }).then(({ json }) => ({ data: json })),
+
+        // simple-rest doesn't handle filters on DELETE route, so we fallback to calling DELETE n times instead
+        deleteMany: (resource, params) =>
+            Promise.all(
+                params.ids.map(id =>
+                    httpClient(`${apiUrl}/${resource}/${id}`, {
+                        method: 'DELETE',
+                        headers: new Headers({
+                            'Content-Type': 'text/plain',
+                        }),
+                    })
+                )
+            ).then(responses => ({
+                data: responses.map(({ json }) => json.id),
+            })),
+    };
+
+    dataProvider.supportAbortSignal = process.env.NODE_ENV === 'production';
```

Comment:
I wouldn't enable it by default in production, otherwise there is  different behavior in dev and prod. Instead, I would document how to enable it.